OrbitClientData/TracepointEventBuffer.cpp

```diff
@@ -89,7 +89,12 @@ uint32_t TracepointEventBuffer::GetNumTracepointsForThreadId(int32_t thread_id)
       return num_total_tracepoints_;
     }
     if (thread_id == -1 /*TODO(b/166238019) Use a proper constant again*/) {
-      return num_total_tracepoints_ - tracepoint_events_.at(kNotTargetProcessThreadId).size();
+      auto const not_target_process_tracepoints_it =
+          tracepoint_events_.find(kNotTargetProcessThreadId);
+      if (not_target_process_tracepoints_it != tracepoint_events_.end()) {
+        return num_total_tracepoints_ - not_target_process_tracepoints_it->second.size();
+      }
+      return num_total_tracepoints_;
     }
   }
```

Comment:
This is not related to this change, but here the mutex is released to then be reacquired in `GetTracepointsOfThread`. While this shouldn't cause issue, it seems wrong to me. This changes the previous behaviour. Why is that, when a `std::recursive_mutex` is used anyway?

Reply:
Changed it.